pandas/tests/arrays/categorical/test_constructors.py

```diff
@@ -560,6 +560,15 @@ def test_from_codes_neither(self):
         with pytest.raises(ValueError, match=msg):
             Categorical.from_codes([0, 1])
 
+    def test_from_codes_with_nullable_int(self):
```

Comment:
can you also test with a NaN in the codes and Int64 (should raise)

Reply:
Done. The error message that ends up getting raised is maybe not the best: `ValueError: cannot convert to '<class 'numpy.int64'>'-dtype NumPy array with missing values. Specify an appropriate 'na_value' for this dtype.`. Would you recommend creating one specific to this method when we have NA values?

Reply:
Yes, I think we should probably catch the error to raise a better one (because the error message now would be really confusing for users)